qrenderdoc/Code/Interface/QRDInterface.h

```diff
@@ -418,6 +418,9 @@ struct ICaptureDialog
   DOCUMENT("Update the current state based on the current remote host, when that changes.");
   virtual void UpdateRemoteHost() = 0;
 
+  DOCUMENT("Called when a remote host is selected.");
+  virtual void OnRemoteHostSwitched() = 0;
```

Comment:
How does this function vary from `UpdateRemoteHost` immediately above it? It seems to try to do the same thing, please don't create duplicate functions.

Reply:
`UpdateRemoteHost` is called in a loop in `MainWindow::messageCheck()` but we need to call this method only when the remote host is actually switched. 
We can use `UpdateRemoteHost` if we remove calling it in `MainWindow::messageCheck()`.

Reply:
If it's being called multiple times then that's a bug, as the comment says it should only be called when the remote host changes.

From looking at the code it seems like it should be inside the block that calls `DisconnectFromRemoteServer();` if the remote server is found to not be running. Fixing the bug is a much better solution than creating a duplicate almost-identical function.